core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/store/dot-container-properties.store.ts

```diff
@@ -1,24 +1,105 @@
+    contentTypes: DotContainerStructure[];
+    apiLink: string;
 }
 
 @Injectable()
 export class DotContainerPropertiesStore extends ComponentStore<DotContainerPropertiesState> {
-    constructor() {
+    constructor(
+        private dotMessageService: DotMessageService,
+        private dotGlobalMessageService: DotGlobalMessageService,
+        private dotContainersService: DotContainersService,
+        private dotHttpErrorManagerService: DotHttpErrorManagerService,
+        private activatedRoute: ActivatedRoute
+    ) {
         super({
-            showPrePostLoopInput: false
+            showPrePostLoopInput: false,
+            isContentTypeVisible: false,
+            isContentTypeButtonEnabled: false,
+            contentTypes: [],
+            container: null,
+            apiLink: ''
         });
+        this.activatedRoute.data
+            .pipe(pluck('container'), take(1))
+            .subscribe((containerEntity: DotContainerEntity) => {
+                if (containerEntity) {
+                    const { container, contentTypes } = containerEntity;
+                    if (container && (container.preLoop || container.postLoop)) {
+                        this.updatePrePostLoopAndContentTypeVisibility({
+                            showPrePostLoopInput: true,
+                            isContentTypeVisible: true,
+                            container: container,
+                            contentTypes: contentTypes ?? []
+                        });
+                    } else {
+                        this.updateContainerState(containerEntity);
+                    }
+
+                    this.updateApiLink(container.identifier);
+                }
+            });
     }
 
-    readonly vm$ = this.select(({ showPrePostLoopInput }: DotContainerPropertiesState) => {
-        return {
-            showPrePostLoopInput
-        };
+    readonly vm$ = this.select((state: DotContainerPropertiesState) => {
+        return state;
     });
 
+    readonly containerAndStructure$ = this.select(
+        ({ container, contentTypes }: DotContainerPropertiesState) => {
+            return {
+                container,
+                contentTypes
+            };
+        }
+    );
+
+    readonly updatePrePostLoopAndContentTypeVisibility = this.updater<{
+        showPrePostLoopInput: boolean;
+        isContentTypeVisible: boolean;
+        container: DotContainer;
+        contentTypes: DotContainerStructure[];
+    }>(
+        (
+            state: DotContainerPropertiesState,
+            {
+                showPrePostLoopInput,
+                isContentTypeVisible,
+                container,
+                contentTypes
+            }: DotContainerPropertiesState
+        ) => {
+            return {
+                ...state,
+                showPrePostLoopInput,
+                isContentTypeVisible,
+                container,
+                contentTypes
+            };
+        }
+    );
```

Comment:
It does not work because we need to update specific values, or we need to change some parameters optional in the `DotContainerPropertiesState` Interface